hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -299,37 +299,80 @@ private RegionLocations addToCache(TableCache tableCache, RegionLocations locs)
     LOG.trace("Try adding {} to cache", locs);
     byte[] startKey = locs.getRegionLocation().getRegion().getStartKey();
     for (;;) {
-      RegionLocations oldLocs = tableCache.cache.putIfAbsent(startKey, locs);
```

Comment:
the changes here are almost all indentation. The only other change I made was three calls to `cleanOverlappingRegions`.